### PR TITLE
Use PODMAN_USERNS environment variable when running as a service

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/11350#issuecomment-1011562526

Also add inspect information about the idmappings if they exists.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: test/e2e/run_userns_test.go [hunk 1]
```diff
@@ -301,5 +301,32 @@ var _ = Describe("Podman UserNS support", func() {
...
+
+		inspect := podmanTest.Podman([]string{"inspect", "--format", "{{ .HostConfig.IDMappings }}", result.OutputToString()})
+		inspect.WaitWithDefaultTimeout()
+		Expect(inspect.OutputToString()).To(Not(Equal("<nil>")))
+
+		if podmanUserusSet {
+			os.Setenv("PODMAN_USERNS", podmanUserns)
+		} else {
+			os.Unsetenv("PODMAN_USERNS")
+		}
```
Comment (vrothberg) [inline test/e2e/run_userns_test.go:327]: Can you move this into a `defer` right after setting it?

**Review by vrothberg — COMMENTED** [review]
Besides the nit, LGTM

**Pull request merged**